akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala

```diff
@@ -126,65 +127,81 @@ object Framing {
   }
 
   private class DelimiterFramingStage(val separatorBytes: ByteString, val maximumLineBytes: Int, val allowTruncation: Boolean)
-    extends PushPullStage[ByteString, ByteString] {
-    private val firstSeparatorByte = separatorBytes.head
-    private var buffer = ByteString.empty
-    private var nextPossibleMatch = 0
+    extends GraphStage[FlowShape[ByteString, ByteString]] {
 
-    override def onPush(chunk: ByteString, ctx: Context[ByteString]): SyncDirective = {
-      buffer ++= chunk
-      doParse(ctx)
-    }
+    val in = Inlet[ByteString]("DelimiterFramingStage.in")
+    val out = Outlet[ByteString]("DelimiterFramingStage.out")
+    override val shape: FlowShape[ByteString, ByteString] = FlowShape(in, out)
```

Comment:
I think setting the name attribute using `override def initialAttributes: Attributes = DefaultAttributes.delimiterFraming` like the stages in `Ops.scala` would be cleanerthan in the factory method, also, a `toString` would be nice.